sdk/benches/ed25519_instructions.rs

```diff
@@ -0,0 +1,69 @@
+#![feature(test)]
+
+extern crate test;
+use {
+    rand0_7::thread_rng,
+    solana_sdk::{
+        ed25519_instruction::new_ed25519_instruction,
+        feature_set::FeatureSet,
+        hash::Hash,
+        signature::{Keypair, Signer},
+        transaction::Transaction,
+    },
+    test::Bencher,
+};
+
+fn create_test_transaction(message_length: u16) -> Transaction {
+    let privkey = ed25519_dalek::Keypair::generate(&mut thread_rng());
+    let message_arr = vec![1u8; message_length as usize];
+    let instruction = new_ed25519_instruction(&privkey, &message_arr);
+    let mint_keypair = Keypair::new();
+
+    Transaction::new_signed_with_payer(
+        &[instruction.clone()],
+        Some(&mint_keypair.pubkey()),
+        &[&mint_keypair],
+        Hash::default(),
+    )
+}
+
+#[bench]
+fn bench_ed25519_len_032(b: &mut Bencher) {
+    let tx = create_test_transaction(32);
+    let feature_set = FeatureSet::all_enabled();
+
+    b.iter(|| {
+        tx.verify_precompiles(&feature_set).unwrap();
```

Comment:
not 100% sure, but could benchmark being optimizing things here because tx is unchanging so the verification of precompiles is always the same? 

Could probably do something along the lines of:
1. create multiple transactions and use a cycling iterator so we can rotate them
2. use black_box so compiler is pessimistic about the value of `tx`

Reply:
using prepared unique txs for benching, the result is about the same